hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/stats/SidecarInstrumentationImpl.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.mono.stats;
+
+import com.hedera.services.stream.proto.SidecarType;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.EnumMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Holds the accumulators needed for the metrics/logging of a sidecar record (corresponding, most
+ * likely, to a record block. There's a separate accumulator for each kind of sidecar record
+ * (bytecode, state change, actions).
+ */
+public record SidecarInstrumentationImpl(
```

Comment:
Is that a style preference only or something more?  Because it has some syntax simplicity + the benefit of immutable fields (which would be _more valuable_ if _non-primitive_ fields were `const` as in C++, I agree).  I'm not much fussed either way, but there's no particular guidance one way or the other, or at least, nothing that's _prescriptive_, in the Java web-references-tutorials-blog-posts-etc that I've yet seen.

Reply:
I don't think, that you will find anything _prescriptive_. The closest I am aware of are the goals and non-goals in the JEP. There it says that records are meant to simplify the definition of data carriers / aggregations of values, but not to "declare a war on boilerplate".

We should probably add a clarifying statement in our coding style guide.